apps/calendar/js/app.js

```diff
@@ -84,15 +83,25 @@ module.exports = {
     this.syncController = new SyncController(this);
     this.serviceController = new ServiceController(this);
     this.errorController = new ErrorController(this);
-    notificationsController.app = this;
-    periodicSyncController.app = this;
+    periodicSyncController.syncController = this.syncController;
 
-    dayObserver.busytimeStore = this.store('Busytime');
-    dayObserver.calendarStore = this.store('Calendar');
-    dayObserver.eventStore = this.store('Event');
     dayObserver.syncController = this.syncController;
     dayObserver.timeController = this.timeController;
 
+    // avoid circular dependencies
+    storeFactory.providerFactory = providerFactory;
+    providerFactory.storeFactory = storeFactory;
+    providerFactory.db = db;
+    providerFactory.app = this;
+    mutations.storeFactory = storeFactory;
+    notificationsController.db = db;
+    notificationsController.storeFactory = storeFactory;
+    injected.app = this;
+    injected.db = db;
+    injected.timeController = this.timeController;
+    injected.providerFactory = providerFactory;
+    injected.storeFactory = storeFactory;
```

Comment:
I think in the end we will need something like the `app` namespace where we can store the references to the main objects and factories to avoid circular dependencies.. the amount of dependency injection is getting out of control and the amount of dependencies on the app.js file is insane.

here I'm calling it `injected` and only used it in few places but maybe we can find a better name and use it everywhere.